packages/presentation/src/communication.ts

```diff
@@ -0,0 +1,267 @@
+    const event: CreateNotificationEvent = {
+      type: EventType.CreateNotification,
+      message,
+      context
+    }
+    await this.connection.sendEvent(event)
+  }
+
+  async removeNotification (message: MessageID, context: ContextID): Promise<void> {
+    const event: RemoveNotificationEvent = {
+      type: EventType.RemoveNotification,
+      message,
+      context
+    }
+    await this.connection.sendEvent(event)
+  }
+
+  async createNotificationContext (card: CardID, lastView?: Date, lastUpdate?: Date): Promise<ContextID> {
+    const event: CreateNotificationContextEvent = {
+      type: EventType.CreateNotificationContext,
+      card,
+      lastView,
+      lastUpdate
+    }
+    const result = await this.connection.sendEvent(event)
+    return (result as CreateNotificationContextResult).id
+  }
+
+  async removeNotificationContext (context: ContextID): Promise<void> {
+    const event: RemoveNotificationContextEvent = {
+      type: EventType.RemoveNotificationContext,
+      context
+    }
+    await this.connection.sendEvent(event)
+  }
+
+  async updateNotificationContext (context: ContextID, update: NotificationContextUpdate): Promise<void> {
+    const event: UpdateNotificationContextEvent = {
+      type: EventType.UpdateNotificationContext,
+      context,
+      update
+    }
+    await this.connection.sendEvent(event)
+  }
+
+  async findMessages (params: FindMessagesParams, queryId?: number): Promise<Message[]> {
+    const rawMessages = await this.connection.findMessages(params, queryId)
+    return rawMessages.map((it: any) => this.toMessage(it))
+  }
+
+  private toMessage (raw: any): Message {
+    return {
+      id: raw.id,
+      card: raw.card,
+      content: raw.content,
+      creator: raw.creator,
+      created: new Date(raw.created),
+      edited: new Date(raw.edited),
+      reactions: raw.reactions.map((it: any) => this.toReaction(it)),
+      attachments: raw.attachments.map((it: any) => this.toAttachment(it))
+    }
+  }
+
+  private toAttachment (raw: any): Attachment {
+    return {
+      message: raw.message,
+      card: raw.card,
+      creator: raw.creator,
+      created: new Date(raw.created)
+    }
+  }
+
+  private toReaction (raw: any): Reaction {
+    return {
+      message: raw.message,
+      reaction: raw.reaction,
+      creator: raw.creator,
+      created: new Date(raw.created)
+    }
+  }
+
+  async findNotificationContexts (
+    params: FindNotificationContextParams,
+    queryId?: number
+  ): Promise<NotificationContext[]> {
+    return await this.connection.findNotificationContexts(params, queryId)
+  }
+
+  async findNotifications (params: FindNotificationsParams, queryId?: number): Promise<Notification[]> {
+    return await this.connection.findNotifications(params, queryId)
+  }
+
+  async unsubscribeQuery (id: number): Promise<void> {
+    await this.connection.unsubscribeQuery(id)
+  }
+
+  close (): void {
+    // do nothing
```

Comment:
Shouldn't the connection be closed here?
`this. connection.close()`

Reply:
Looks like communication client uses connection from other client, so it shouldn't be closed here